asim: improve data driven test output

wenyihu6 opened the pull request:
**asim: improve data driven test output**

This commit prints simulation state for gen_load, gen_ranges, and gen_cluster
before the simulation run to make these info more explicit.

Epic: none
Release note: none

---

**asim: record cpu utilization in StoreMetrics**

Previously, StoreMetrics didn't include CPU utilization, making it hard to
evaluate the simulation setup. This commit adds CPU utilization to the store
metrics. Note that due to the lack of node-level metrics, the same CPU capacity
will be repeated for stores on the same node.

Epic: none
Release note: none

wenyihu6 committed:
asim: improve data driven test output

This commit prints simulation state for gen_load, gen_ranges, and gen_cluster
before the simulation run to make these info more explicit.

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/150511)
<!-- Reviewable:end -->

wenyihu6 committed:
asim: record cpu utilization in StoreMetrics

Previously, StoreMetrics didn't include CPU utilization, making it hard to
evaluate the simulation setup. This commit adds CPU utilization to the store
metrics. Note that due to the lack of node-level metrics, the same CPU capacity
will be repeated for stores on the same node.

Epic: none
Release note: none

tbg reviewed (changes requested):
Thank you for doing this! I requested changes because some of the new functionality seems to be unused, so I'd prefer you either adopt it or move it into a new PR that adopts it. (Mild preference to just adopt it, since that seems easier for you).

> Reviewed 41 of 41 files at r1, 23 of 23 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/150511#-:-OVc1hqBDkmcYXHxU4SP:btt3v4a)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[`pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go` line 556 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/150511#-OVc0gtbChJQyQKP3q19:-OVc0gtbChJQyQKP3q1A:b-68i7yj) ([raw file](https://github.com/cockroachdb/cockroach/blob/3a25ea72b60bc0148d1500ae8cf2390fe9936963/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go#L556)):*
> ```Go
> 			case "print":
> 				var state string = "cluster"
> 				scanIfExists(t, d, "state", &state)
> ```

Is this unused? +1 on making `print` more useful, it dumps giant walls of text right now. But then please adopt it, too :-)
___
*[`pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go` line 581 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/150511#-OVc0VQBAsOLYPTJKKVw:-OVc0VQBAsOLYPTJKKVx:bvrldwj) ([raw file](https://github.com/cockroachdb/cockroach/blob/3a25ea72b60bc0148d1500ae8cf2390fe9936963/pkg/kv/kvserver/asim/tests/datadriven_simulation_test.go#L581)):*
> ```Go
> 				scanIfExists(t, d, "height", &height)
> 				scanIfExists(t, d, "width", &width)
> 				scanIfExists(t, d, "format", &format)
> ```

This seems unused so far? I was also planning to make the datadriven files "test specs" (i.e. their direct inlined datadriven output would be independent of the allocator used) and generating all graphs automatically, which clashes with this change.


<!-- Sent from Reviewable.io -->
